worker/worker.go

```diff
@@ -190,18 +191,30 @@ func (w *worker) pollPlayers(ctx context.Context) {
 }
 
 func (w *worker) checkPlayer(ctx context.Context, p api.GetPlayerResponse) {
-	// If player is not spawned, reset first coordinate tracking and return
-	if !p.Position.IsSpawned() {
-		w.firstCoordSkipped.Delete(p.Id)
-		w.outsidePlayers.Delete(p.Id)
-		return
-	}
+// Skip players who are not spawned or have invalid position (0,0,0)
+    	if !p.Position.IsSpawned() || (p.Position.X == 0 && p.Position.Y == 0 && p.Position.Z == 0) {
```

Comment:
`IsSpawned()` is already checking for the 0,0,0 vector. No need to do that again :)